tests/db/utils_test.py

```diff
@@ -380,3 +387,235 @@ def test_unicode_as_bytes(self):
+
+        ctx = gpgme.Context()
+        search_dir = os.path.join(os.path.dirname(__file__), '../static/gpg-keys')
+        for each in os.listdir(search_dir):
+            if os.path.splitext(each)[1] == '.gpg':
+                with open(os.path.join(search_dir, each)) as f:
+                    ctx.import_(f)
+
+    def test_erase_alot_header_signature_valid(self):
+        """Alot uses special headers for passing certain kinds of information,
+        it's important that information isn't passed in from the original
+        message as a way to trick the user.
+        """
+        m = email.message.Message()
+        m.add_header(utils.X_SIGNATURE_VALID_HEADER, 'Bad')
+        message = utils.message_from_file(io.BytesIO(m.as_string()))
+        self.assertIs(message.get(utils.X_SIGNATURE_VALID_HEADER), None)
+
+    def test_erase_alot_header_message(self):
+        m = email.message.Message()
+        m.add_header(utils.X_SIGNATURE_MESSAGE_HEADER, 'Bad')
+        message = utils.message_from_file(io.BytesIO(m.as_string()))
+        self.assertIs(message.get(utils.X_SIGNATURE_MESSAGE_HEADER), None)
+
+    def test_plain_mail(self):
+        m = email.mime.text.MIMEText(u'This is some text', 'plain', 'utf-8')
+        m['Subject'] = 'test'
+        m['From'] = 'me'
+        m['To'] = 'Nobody'
+        message = utils.message_from_file(io.BytesIO(m.as_string()))
+        self.assertEqual(message.get_payload(), 'This is some text')
+
+    @staticmethod
+    def _make_signed():
+        """Create a signed message that is multipart/signed."""
+        text = 'This is some text'
+        t = email.mime.text.MIMEText(text, 'plain', 'utf-8')
+        _, sig = crypto.detached_signature_for(helper.email_as_string(t))
+        s = email.mime.application.MIMEApplication(
+            sig, 'pgp-signature', email.encoders.encode_7or8bit)
+        m = email.mime.multipart.MIMEMultipart('signed', None, [t, s])
+        m.set_param('protocol', 'application/pgp-signature')
+        m.set_param('micalg', 'pgp-sha256')
+        return m
+
+    def test_signed_headers_included(self):
+        """Headers are added to the message."""
+        m = self._make_signed()
+        m = utils.message_from_file(io.BytesIO(m.as_string()))
+        self.assertIn(utils.X_SIGNATURE_VALID_HEADER, m)
+        self.assertIn(utils.X_SIGNATURE_MESSAGE_HEADER, m)
+
+    def test_signed_valid(self):
+        """Test that the signature is valid."""
+        m = self._make_signed()
+        m = utils.message_from_file(io.BytesIO(m.as_string()))
+        self.assertEqual(m[utils.X_SIGNATURE_VALID_HEADER], 'True')
+
+    def test_signed_correct_from(self):
+        """Test that the signature is valid."""
+        m = self._make_signed()
+        m = utils.message_from_file(io.BytesIO(m.as_string()))
+        # Don't test for valid/invalid since that might change
+        self.assertIn('ambig <ambig@example.com>', m[utils.X_SIGNATURE_MESSAGE_HEADER])
```

Comment:
It's from the keys that are stored as static data with the tests. I'm not sure how to make this cleaner, but I'm opened to ideas.